lib/ui/painting/multi_frame_codec.cc

```diff
@@ -115,17 +115,17 @@ sk_sp<DlImage> MultiFrameCodec::State::GetNextFrameImage(
           << "Frame " << nextFrameIndex_ << " depends on frame "
           << requiredFrameIndex
           << " and no required frames are cached. Using blank slate instead.";
+      return nullptr;
     } else if (lastRequiredFrameIndex_ != requiredFrameIndex) {
       FML_DLOG(INFO) << "Required frame " << requiredFrameIndex
                      << " is not cached. Using blank slate instead.";
-    } else {
-      // Copy the previous frame's output buffer into the current frame as the
-      // starting point.
-      if (lastRequiredFrame_->getPixels() &&
-          CopyToBitmap(&bitmap, lastRequiredFrame_->colorType(),
-                       *lastRequiredFrame_)) {
-        prior_frame_index = requiredFrameIndex;
-      }
+    }
+    // Copy the previous frame's output buffer into the current frame as the
+    // starting point.
+    if (lastRequiredFrame_->getPixels() &&
```

Comment:
I _think_ adding `requiredFrameIndex != SkCodec::kNoFrame` to the conditional would keep both fixes working.

Reply:
I did this a bit differently by making the condition below that updates the "keep" frame every time once we first encounter a keep frame.

I think this is fine because Skia's documentation on this parameter says that the last required frame is the _earliest_ frame, and you can use any frame from that one to the last one. See https://github.com/google/skia/blob/7b18d6c5c676abcb274a8dc6b2896c8c2f667a63/include/codec/SkCodec.h#L640